narwhal/primary/src/synchronizer.rs

```diff
@@ -252,18 +327,20 @@ impl Synchronizer {
         certificate: Certificate,
         network: &Network,
     ) -> DagResult<()> {
-        self.process_certificate_internal(certificate, network, true)
+        self.process_certificate_internal(certificate, network, true, true)
             .await
     }
 
-    /// Tries to accept a certificate that is already sanitized.
-    // TODO: produce a different type after sanitize, e.g. VerifiedCertificate.
-    pub async fn try_accept_sanitized_certificate(
+    /// Tries to accept a certificate from certificate fetcher.
+    /// Fetched certificates are already sanitized in parallel.
+    /// And in most cases, they have no missing parents. So it is useful to always check parents
```

Comment:
Comment is a little confusing because you are passing `expect_missing_parents` as false but are saying it is always useful to check parents. Maybe change it to something like 

"And in most cases we expect there to be no missing parents. However we still always check parents in any case while processing certificates and use this information about fetched certificates to validate suspended certificates state."

Reply:
Good point, I reworded this and also renamed `expect_missing_parents` to `early_suspend`. Hope this is clearer.